main.py

```diff
@@ -1,27 +1,73 @@
-from flask import Flask, render_template
+from flask import Flask, render_template, request, send_from_directory
 import yaml
 from yaml.loader import SafeLoader
+from werkzeug.utils import secure_filename
+import tempfile
+import os
+import subprocess as sp
+
 
 app = Flask("SINTEF MIA Catalogue")
 APP_DICT = dict()
 
+
 @app.route("/")
 def main():
     ctx = load_catalogue()
     for app in ctx['apps']:
         APP_DICT[app['name']] = app
     return render_template("catalogue.jinja2", title="Catalogue", **ctx)   # By default, Flask expects your templates in a templates/ directory
 
+
 @app.route("/application/<string:app_name>")
 def application(app_name):
     return render_template("{}.jinja2".format(app_name.lower()), **{'app_name': app_name.capitalize()})
 
 
-# @app.route("process/<string:app_name>", methods=['GET', 'POST'])
-# def process(app_name):
-#     if request.method == 'POST':
-#         # Forward the uploaded data to the corresponding service
-#         APP_DICT[app_name].
+@app.route("/")
```

Comment:
You can use the same route i.e., "/", for multiple functions depending on the communication method used, e.g., GET (default), POST, etc.
Linking the same route with the same method, to different functions is not advised.

I would suggest merging the functions "application" and "process", and use the same template (or collection of templates) for all the services . These can be adapted, and changes e.g., show/hide the spinning circle, based on the status of the task.

Reply:
Yh, I agree. I was just playing around with stuff here.